tests/accessor/host_accessor_constructors.h

```diff
@@ -0,0 +1,183 @@
+  auto offset = sycl::id<Dimension>();
+  auto tagT = get_tag<AccessMode>();
+
+  auto section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, "TagT deduction from buffer constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [tagT](sycl::buffer<DataT, Dimension> data_buf) {
+      return sycl::host_accessor<DataT, Dimension, AccessMode>(data_buf,
+                                                                 tagT);
+    };
+    check_common_constructor<AccType, DataT, Dimension, AccessMode>(
+        get_acc_functor);
+  }
+
+  section_name = get_section_name<Dimension>(
+      type_name, access_mode_name,
+      "TagT deduction from buffer and range constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [r, tagT](sycl::buffer<DataT, Dimension> data_buf) {
+      return sycl::host_accessor(data_buf, r, tagT);
+    };
+    check_common_constructor<AccType, DataT, Dimension, AccessMode>(
+        get_acc_functor);
+  }
+
+  section_name =
+      get_section_name<Dimension>(type_name, access_mode_name,
+                                   "TagT deduction from buffer,range and "
+                                   "offset constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = [r, offset,
+                            tagT](sycl::buffer<DataT, Dimension> data_buf) {
+      return sycl::host_accessor(data_buf, r, offset, tagT);
+    };
+    check_common_constructor<AccType, DataT, Dimension, AccessMode>(
+        get_acc_functor);
+  }
+}
+
+template <typename T, typename AccessModeT, typename DimensionT>
+class run_tests_constructors {
+  static constexpr sycl::access_mode AccessMode = AccessModeT::value;
+  static constexpr int Dimension = DimensionT::value;
+
+ public:
+  void operator()(const std::string& type_name,
+                  const std::string& access_mode_name) {
+    test_zero_dimension_buffer_constructor<T, AccessMode>(access_mode_name,
+                                                           type_name);
+    test_default_constructor<T, Dimension, AccessMode>(access_mode_name,
+                                                         type_name);
+    test_common_buffer_constructors<T, Dimension, AccessMode>(
+        access_mode_name, type_name);
+    test_common_buffer_constructors_tag_t_deduction<T, Dimension, AccessMode>(
+        access_mode_name, type_name);
+  }
+};
+
+template <typename T>
+class run_host_constructors_test {
+ public:
+  void operator()(const std::string& type_name) {
+    // Type packs instances have to be const, otherwise for_all_combination will
+    // not compile
+    const auto access_modes = get_access_modes();
+    const auto dimensions = get_dimensions();
+    const auto cur_type =
+        named_type_pack<T>::generate(type_name_string<T>::get(type_name));
+
+    for_all_combinations<run_tests_constructors>(cur_type, access_modes,
+                                                 dimensions);
```

Comment:
```suggestion
    for_all_combinations<run_tests_constructors, T>(access_modes, dimensions, type_name);
```
Will it work in a such way or we have a bottleneck here?

Reply:
Looks like we need to update `for_all_combinations` in order to do this trick.

Reply:
Upd: I was wrong. This actually works.